addons/purchase/purchase.py

```diff
@@ -736,19 +735,19 @@ def propagate_cancels(self):
 
         return result
 
-    @api.model
-    def _run(self, procurement):
-        if procurement.rule_id and procurement.rule_id.action == 'buy':
-            return procurement.make_po()
-        return super(ProcurementOrder, self)._run(procurement)
+    @api.multi
+    def _run(self):
+        if self.rule_id and self.rule_id.action == 'buy':
```

Comment:
Use self.ensure_one() to verify that the self contains only one record or make the method to work on multiple records.